src/main/java/org/sagebionetworks/web/client/FeatureFlagConfigFactory.java

```diff
@@ -0,0 +1,29 @@
+package org.sagebionetworks.web.client;
+
+import com.google.gwt.json.client.JSONObject;
+import com.google.gwt.json.client.JSONParser;
+import com.google.gwt.json.client.JSONValue;
+import com.google.inject.Inject;
+
+public class FeatureFlagConfigFactory {
+
+  PortalGinInjector ginInjector;
+  private JSONObject config;
+
+  @Inject
+  public FeatureFlagConfigFactory(PortalGinInjector ginInjector) {
+    this.ginInjector = ginInjector;
+  }
+
+  public void create(String json) {
+    JSONValue parsed = JSONParser.parseStrict(json);
+    config = parsed.isObject();
+  }
+
+  public boolean isFeatureEnabled(String featureName) {
+    return (
+      DisplayUtils.isInTestWebsite(ginInjector.getCookieProvider()) ||
+      config.get(featureName).isBoolean().booleanValue()
+    );
+  }
```

Comment:
If you want to keep the Factory pattern (so you can dynamically create the class using a config), split this into two classes:

`FeatureFlagConfigFactory` has (at least) one method, `FeatureFlagConfig create(String jsonString)`.

`FeatureFlag` has (at least) one method, `boolean isFeatureEnabled(String featureName)`

Reply:
`FeatureFlagConfig(String jsonObject)`


Is it possible to get the injected cookie provider in this provides method?
```java
@Provides 
FeatureFlagConfig provideFeatureFlagConfig() {
  cookieProvider.getCookie(WebConstants.PORTAL_FEATURE_FLAG)
}

```